packages/compass-data-modeling/src/utils/schema-traversal.tsx

```diff
@@ -165,3 +165,118 @@ export const getFieldFromSchema = ({
     parentFieldPath: [...parentFieldPath, nextInPath],
   });
 };
+
+type MutationParameters = {
+  update: 'removeField' | 'renameField';
+  newFieldName?: string;
+};
+
+const getMutatedSchema = ({
+  schema,
+  fieldName,
+  newFieldName,
+  update,
+}: {
+  schema: MongoDBJSONSchema;
+  fieldName: string;
+} & MutationParameters): MongoDBJSONSchema => {
+  switch (update) {
+    case 'removeField': {
+      if (!schema.properties || !schema.properties[fieldName])
+        throw new Error('Field to remove does not exist');
+      return {
+        ...schema,
+        properties: Object.fromEntries(
+          Object.entries(schema.properties).filter(([key]) => key !== fieldName)
+        ),
+      };
+    }
+    case 'renameField': {
+      if (!schema.properties || !schema.properties[fieldName])
+        throw new Error('Field to rename does not exist');
+      if (!newFieldName)
+        throw new Error('New field name is required for the rename operation');
```

Comment:
Don't mind the runtime checks, but shouldn't this just be enforced by types? If we're renaming a field, new name should be required, right?